src/main/java/chalmers/pimp/service/ImageImportService.java

```diff
@@ -29,4 +32,15 @@ public static Image importImage(File file) throws IOException {
       throw new IOException("Failed to load the image at: " + file);
     }
   }
+
+  /**
+   * Returns the given file's filename (without extensions).
+   *
+   * @param file the file to have it's name returned.
+   * @return the name of the supplied file.
+   */
+  public static String getFileNameFromFile(File file) {
+    List<String> temp = new ArrayList<>(Arrays.asList(file.getName().split("\\.")));
+    return file.getName().replace("." + temp.get(temp.size() - 1), "");
+  }
```

Comment:
Looks good but should this really be in ImageImportService? (Also what happens when the supplied file is null?)

Reply:
Could make new class for this, but the whole method is pretty dependant on there being a "imported file". Just a matter of preference, will do as Merge-Master commands :neckbeard:

Reply:
It’s just that a File instance doesn’t have to point to an image